Tests/ClientTests/Ecosia/EcosiaInstallTypeTests.swift

```diff
@@ -65,6 +67,53 @@ final class EcosiaInstallTypeTests: XCTestCase {
     }
 }
 
+extension EcosiaInstallTypeTests {
+    
+    // Test evaluating install type and version for a fresh install with firstTime=true
+    func testEvaluateFreshInstallType_WithFirstTime_And_VersionProvider() {
+        User.shared.firstTime = true
+        let versionProvider = MockAppVersionInfoProvider(mockedAppVersion: "1.0.0")
+        EcosiaInstallType.evaluateCurrentEcosiaInstallTypeWithVersionProvider(versionProvider)
+        
+        XCTAssertEqual(EcosiaInstallType.get(), .fresh)
+        XCTAssertEqual(EcosiaInstallType.persistedCurrentVersion(), "1.0.0")
+    }
+
+    // Test evaluating install type and version for a fresh install with firstTime=false
+    func testEvaluateFreshInstallType_WithFirstTimeFalse_And_VersionProvider() {
+        User.shared.firstTime = false
+        let versionProvider = MockAppVersionInfoProvider(mockedAppVersion: "1.0.0")
+        EcosiaInstallType.evaluateCurrentEcosiaInstallTypeWithVersionProvider(versionProvider)
+        
+        XCTAssertEqual(EcosiaInstallType.get(), .upgrade)
+        XCTAssertEqual(EcosiaInstallType.persistedCurrentVersion(), "1.0.0")
+    }
+
+    // Test evaluating install type and version for an upgrade with firstTime=true
+    func testEvaluateUpgradeInstallType_WithFirstTime_And_VersionProvider() {
```

Comment:
I guess this name also need to be updated, since it is now evaluating fresh install and not upgrade as it is forced by `User.shared.firstTime`.

Reply:
Same when I swapped them 😅  thanks!